src/Eloquent/Collection.php

```diff
@@ -49,4 +50,59 @@ public function toTree($childrenRelation = 'children')
 
         return $tree;
     }
+
+    /**
+     * Load parent/anscestor relations already present in the tree.
+     *
+     * @return static<int, TModel>
+     */
+    public function loadTreePathRelations()
```

Comment:
```suggestion
    public function loadTreePathRelations(): self
```

I think we can safely define return types on new code